compilers/concrete-compiler/compiler/include/concretelang/Common/Keysets.h

```diff
@@ -51,10 +52,22 @@ struct Keyset {
 
   Keyset(){};
 
-  /// Generates a fresh keyset from infos.
+  /// @brief Generates a keyset from infos.
+  ///
+  /// This can be a fresh keyset if no key is specified in `lweSecretKeys`.
+  /// Otherwise those keys are set first, then the rest of the key will be
+  /// generated.
+  ///
+  /// @param info
+  /// @param secretCsprng
+  /// @param encryptionCsprng
+  /// @param lweSecretKeys secret keys to initialize the keyset with
   Keyset(const Message<concreteprotocol::KeysetInfo> &info,
          concretelang::csprng::SecretCSPRNG &secretCsprng,
-         csprng::EncryptionCSPRNG &encryptionCsprng);
+         csprng::EncryptionCSPRNG &encryptionCsprng,
+         std::map<uint32_t, LweSecretKey> lweSecretKeys =
+             std::map<uint32_t, LweSecretKey>());
+
```

Comment:
Can't that be a separate functions ? I think it would be better to have separate entry points for that.

Reply:
The default thing make it possible to use it in two different ways. So you can basically call it like `Keyset(info, secretCsprng, encryptionCsprng)` or `Keyset(info, secretCsprng, encryptionCsprng, lweSKs)`. Or is there any other benefit of having completely separate functions?

Reply:
In my opinion the benefit is to better separate logically different code paths, which leads to simpler code in the function body. One default-arg like that does not do much arm, but as we add more, the logic in the scope of the function gets exponentially difficult to grasph. I just find it a generally good practice to avoid doing it.

It is just an opinion, feel free to ignore if that does not make sense to you :)